src/backend/opencl/platform.cpp

```diff
@@ -507,7 +499,7 @@ void setDeviceContext(cl_device_id dev, cl_context ctx) {
 
     common::lock_guard_t lock(devMngr.deviceMutex);
 
-    const int dCount = devMngr.mDevices.size();
+    const size_t dCount = devMngr.mDevices.size();
     for (int i = 0; i < dCount; ++i) {
```

Comment:
```suggestion
    for (size_t i = 0; i < dCount; ++i) {
```

Reply:
that would avoid the warning of signed and unsigned type comparison

Reply:
When trying, I get a new warning at setActiveContext(i).
As result, I decided to stay at the original int and to add a static_cast<int> at the first assignment instead.  It also solves the warning you have, and which MSVC does not report.

Reply:
You mean, you directly did this `const int dCount = static_cast<int>(devMngr.mDevices.size());` ?

Reply:
That's right.  Gives this a problem somewhere else?

Reply:
Nope, sure do it.